scripts/build/build/__init__.py

```diff
@@ -39,7 +39,8 @@ def __init__(self, runner, repository_path:str, output_prefix:str):
 
   def SetupBuilders(self, platforms: Sequence[Platform],
                     boards: Sequence[Board],
-                    applications: Sequence[Application]):
+                    applications: Sequence[Application],
+                    **kwargs):
```

Comment:
the create_flash_bundle seems to be a property that never changes, it seems it  would be better to pre-set this.
**kargs is not sufficiently type safe - accepting any key  argument does not seem like something we require:

suggestion:
   - in the constructor, set a `self.create_flash_bundles = false`
   - Provide as `enable_flash_bundles(self, value)` method or allow direct setting to = true that you do on the obj after creation and before calling SetupBuilders()

Alternatively you can also pass it as a constructor argument, however I am concerned that increasing method argument count is not ideal. We are forced to do this as command line arguments in `click` however in other places I would prefer to keep the method argument focused.

Reply:
Done, and the factory still uses "enable_flashbundle" as an argument.

I think it is unlikely to change the value of "enable_flashbundle" after it is constructed regarding other building parameters.

We can consider using a dataclass as config like we do in C++ when we are adding more parameters to builder class.